### PR TITLE
introduce 'everest advance'

### PR DESCRIPTION
**Pull request by mtzguido**
Does something like this already exists? It's a shortcut to updating every, or the named subprojects to master.

Would need a few tweaks before merging, but curious how the CI machine does this? Have I just missed the relevant command?

Here's an example of advancing F*:
```
./everest FStar advance
# Switching to the everest directory ... now in /home/guido/r/everest


Advancing FStar
~/r/everest/FStar ~/r/everest
Already on 'master'
Your branch is behind 'origin/master' by 10 commits, and can be fast-forwarded.
  (use "git pull" to update your local branch)
HEAD is now at 15fe9e9322 Merge branch 'master' of github:FStarLang/FStar
~/r/everest
```

I suppose it's similar to `./everest forall git pull`?

### CODE REVIEW

File: everest [hunk 1]
```diff
@@ -894,6 +900,24 @@ do_reset ()
...
+    echo
+    blue "Advancing $r"
+    check_subp_exists "$r"
+
+    pushd $r
+    branch=${branches[$r]}
+
+    git checkout -f $branch
+    git fetch
+    git reset --hard origin/$branch
```
Comment (msprotz) [inline everest:915]: maybe ask for confirmation if there's local changes? this may destruct some users' working states and surprise them

**Comment by mtzguido** [comment]
Good point, thanks! Just added a very conservative check for that. I reused `--yes` to skip the check, let me know if a separate `-f/--force` would be better.

Also a tweak to the sorting `hashes.sh` logic, apparently case-sensitivity varies across systems (or locales maybe).

**Comment by mtzguido** [comment]
Ping, totally non urgent but would come in handy for me.

**Comment by msprotz** [comment]
Sorry, I thought you'd merge after I approved. All looks good, thanks!!

**Pull request merged**